Block/Checkout/Success.php

```diff
@@ -116,6 +141,50 @@ public function getAction()
         return json_encode($this->getOrder()->getPayment()->getAdditionalInformation('action'));
     }
 
+    public function showAdyenGiving()
+    {
+        return $this->adyenGivingEnabled() && $this->hasDonationToken();
+    }
+
+    public function adyenGivingEnabled(): bool
+    {
+        return (bool) $this->configHelper->adyenGivingEnabled($this->storeManager->getStore()->getId());
+    }
+
+    public function hasDonationToken()
+    {
+        return $this->getDonationToken() && 'null' !== $this->getDonationToken();
+    }
+
+    public function getDonationToken()
+    {
+        return json_encode($this->getOrder()->getPayment()->getAdditionalInformation('donationToken'));
+    }
+
+    public function getDonationComponentConfiguration(): array
+    {
+        $storeId = $this->storeManager->getStore()->getId();
+        $imageBaseUrl = $this->storeManager->getStore()->getBaseUrl(\Magento\Framework\UrlInterface::URL_TYPE_MEDIA).'adyen/';
+        $donationAmounts = explode(',', $this->configHelper->getAdyenGivingDonationAmounts($storeId));
+        $donationAmounts = array_map(function ($amount) {
+            return $this->adyenHelper->formatAmount($amount, $this->getOrder()->getOrderCurrencyCode());
```

Comment:
I think this is conflicting since we're requesting the suggested amounts in the config field in EUR. `getAdyenGivingDonationAmounts` will then return amounts in EUR and the order could be in a different currency. How about if we create a helper method that is fed the amounts and the order currency and converts them to the order currency? Similar to how is done in `Model/Config/Backend/DonationAmounts.php`

Reply:
Since this would require a conversion we can park it for now and inform in the config help tooltip that the order currency will be used.